### PR TITLE
Test Android Breakpad Minidumps

### PR DESCRIPTION
**Pull request by Swatinem**
## :scroll: Description

Just for testing android breakpad

## :pencil: Checklist

- [ ] I reviewed the submitted code
- [ ] I added tests to verify the changes
- [ ] I updated the docs if needed
- [ ] No breaking changes

### CODE REVIEW

File: sentry/src/main/java/io/sentry/OutboxSender.java [hunk 1]
```diff
@@ -107,89 +108,97 @@ private void processEnvelope(final @NotNull SentryEnvelope envelope, final @Null
...
+//        ((Resettable) hint).reset();
+//      }
+//    }
+
+    // event and minidump need to be within the same envelope, so right now we are not enriching
+    // the event as a hacky solution, but event is gonna ne symbolicated using the minidump
+    hub.captureEnvelope(envelope, hint);
+    if (!waitFlush(hint)) {
+      return;
     }
```
Comment (marandaneto) [inline sentry/src/main/java/io/sentry/OutboxSender.java:200]: we'd need to special case this if we go ahead with mini dumps, as we need to enrich the event but still sending all the left items within the same envelope

File: sentry/src/main/java/io/sentry/SentryEnvelopeItemHeader.java [hunk 2]
```diff
@@ -14,6 +14,7 @@
   private final @NotNull SentryItemType type;
   private final int length;
   @Nullable private final Callable<Integer> getLength;
+  private final @Nullable String attachmentType;
```
Comment (marandaneto) [inline sentry/src/main/java/io/sentry/SentryEnvelopeItemHeader.java:17]: as attachmentType had a default value on backend and Android didn't support any other than the default one, we didn't add it, now we need to as values come from 3rd party SDKs eg sentry-native

**Comment by marandaneto** [comment]
@Swatinem can we close this?

**Comment by Swatinem** [comment]
@marandaneto lets keep this open for now, I would love to be able to play around more with breakpad on Android.

Half of the experiment is adding support for passing through the special minidump attachment, I guess that would be interesting to do independently in a separate PR?

**Comment by marandaneto** [comment]
> @marandaneto lets keep this open for now, I would love to be able to play around more with breakpad on Android.
> 
> Half of the experiment is adding support for passing through the special minidump attachment, I guess that would be interesting to do independently in a separate PR?

attachmentType is now preserved but event and minidump would still be sent separated which is still a problem as it's not going to symbolicated.
I will open a separated issue for that

**Comment by romtsn** [comment]
@Swatinem can we close this now? looks stale for a while

**Comment by Swatinem** [comment]
I don’t think it hurts keeping the branch around. And it might also be a good idea to properly implement this. I know of one customer who wanted to have proper minidumps on Android, with all the different threads in it.

**Comment by romtsn** [comment]
> I don’t think it hurts keeping the branch around. And it might also be a good idea to properly implement this. I know of one customer who wanted to have proper minidumps on Android, with all the different threads in it.

let's keep the branch around but close the PR since it's just noise in the ever-growing list of PRs, we can always reopen later

**Pull request closed without merging**